udata/search/query.py

```diff
@@ -63,14 +66,19 @@ def iter(self):
         return SearchIterator(self, result)
 
     def get_body(self):
-        body = {
-            'filter': self.get_filter(),
-            'from': (self.page - 1) * self.page_size,
-            'size': self.page_size,
-            'sort': self.get_sort(),
-            'aggregations': self.get_aggregations(),
-            'fields': [],  # Only returns IDs
-        }
+        qs = Search(using=es.client, index=es.index_name)
+        # Sorting
+        qs = self.get_sort(qs)
+        # Pagination
+        start = (self.page - 1) * self.page_size
+        qs = qs[start:start + self.page_size]
+        # Aggregations
+        qs = self.build_aggregations(qs)
+        # don't return any fields, just the metadata
+        qs = qs.fields([])
+
+        body = qs.to_dict()
```

Comment:
Instead of the `to_dict`/`from_dict`, we can use the [update_from_dict](http://elasticsearch-dsl.readthedocs.io/en/latest/search_dsl.html#serialization-and-deserialization) method here to optionally add scoring functions.